### PR TITLE
Added tests requirs grad and initialization

### PR DESCRIPTION
**Pull request by marii-moe**
Added tests for requires_grad. Also added test for apply_init because it tested a path through requires grad.

### CODE REVIEW

File: tests/test_torch_core.py [hunk 1]
```diff
@@ -19,3 +21,31 @@ def test_tensor_with_tensor():
...
+
+def test_requires_grad():
+    m = simple_cnn(b)
+    assert requires_grad(m) == True
+
+def test_requires_grad_set():
+    m = simple_cnn(b)
+    requires_grad(m,False)
+    disj = False
+    for p in m.parameters():
```
Comment (sgugger) [inline tests/test_torch_core.py]: Those little for loops fit best in one single line. Can you edit your PR accordingly?

**Comment by jph00** [comment]
Actually you should be able to do it without a for loop using something like `torch.all()`.

Please either name your tests with a full name of exactly what's being tested, or add a 2nd string argument to have your assertions explain exactly what they're testing.

**Comment by marii-moe** [comment]
> Actually you should be able to do it without a for loop using something like `torch.all()`.
I spent a bit of time checking for this myself, and was not able to find anything of this nature. I will implement your other suggestions though.

**Comment by marii-moe** [comment]
Latest commit should have fixed above mentioned issues.

**Pull request merged**